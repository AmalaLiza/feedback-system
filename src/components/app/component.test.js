import React from "react";
import { shallow } from "enzyme";
import App from "./component";
import renderer from "react-test-renderer";
import { HashRouter } from "react-router-dom";
import configureStore from "../../store";
import { Provider } from "react-redux";

const mockStore = configureStore();

describe("App Component Test Suite", () => {
	it("renders correctly", () => {
		const tree = renderer
			.create(
				<Provider store={mockStore}>
					<HashRouter>
						<App />
					</HashRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
