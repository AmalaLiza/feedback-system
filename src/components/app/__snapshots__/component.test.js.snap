// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test Suite renders correctly 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    GIVE US FEEDBACK
  </h1>
  <div
    className="container"
  >
    <label
      className="label"
      htmlFor="Name"
    >
      Name
    </label>
    <input
      className="input"
      id="Name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="input"
      value=""
    />
  </div>
  <div
    className="container"
  >
    <label
      className="label"
      htmlFor="Email Id"
    >
      Email Id
    </label>
    <input
      className="input"
      id="Email Id"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="commentInput container"
  >
    <label
      className="label"
      htmlFor="Comments"
    >
      Comments
    </label>
    <input
      className="input"
      id="Comments"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="comment"
      value=""
    />
  </div>
  <label
    className="label"
  >
    Overall Rating
  </label>
  <span
    className="star rating"
    onMouseLeave={[Function]}
    style={
      Object {
        "direction": "ltr",
        "display": "inline-block",
      }
    }
  >
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
    <span
      onClick={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          className="fa fa-star-o fa-2x"
        />
      </span>
      <span
        style={
          Object {
            "display": "inline-block",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "top": 0,
            "width": "0%",
          }
        }
      >
        <span
          className="fa fa-star fa-2x"
        />
      </span>
    </span>
  </span>
  <button
    className="button button"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
